### ISSUE TITLE
Update `WF-0 fetch metadata file` to incorporate metadata sheet versioning rules

### ISSUE THREAD

**Issue by AishaHassen** [OpenFn/openfn-lime-pilot#64]
## Background, context, and business value

We have added a column named `OMRS Form Version` and `Active` in the [`omrs-form-metadata` sheet](https://openfnorg.sharepoint.com/:x:/r/sites/openfn-msf-test/_layouts/15/Doc.aspx?sourcedoc=%7B0FCD6FD8-9541-4CF6-850A-958B37C4044E%7D&file=LIME%20EMR%20-%20Iraq%20Metadata%20-%20Release%201.xlsx&action=default&mobileredirect=true).
![Image](https://github.com/user-attachments/assets/a7b0e8d3-8edc-4fa6-b908-4b710cf8d762)

This `omrs-form-metadata` has already been added to the collection.json file as per this issue #53 and can be accessed as `metadata-json["OMRS Form Version"]`. This returns an array of OMRS sheets and their corresponding version number:
```
​​[
    {
        "OMRS form name": "F01-MHPSS Baseline",
        "OMRS form.uuid": "6a3e1e0e-dd13-3465-b8f5-ee2d42691fe5",
        "DHIS2 program ID": "w9MSPn5oSqp",
        "DHIS2 programStage ID": "MdTtRixaC1B",
        "OMRS Form Version": "v1"
    },
    {
        "OMRS form name": "F03-mhGAP Baseline",
        "OMRS form.uuid": "82db23a1-4eb1-3f3c-bb65-b7ebfe95b19b",
        "DHIS2 program ID": "w9MSPn5oSqp",
        "DHIS2 programStage ID": "EZJ9FsNau7Q",
        "OMRS Form Version": "v1"
    }]

```

## The specific request, in as few words as possible

1. We will use the **Active** sheet to determine which sheets to fetch or not. 
     a. If the Active status for a sheet is `TRUE`, that means it should be synced(if the version number has changed)
     b. If the Active status for a sheet is `FALSE`, that means it might have been deleted. We need to make sure that it is also removed from the `collection.json` file.

2. Make sure that for each "Active" sheet, refer to the column `OMRS form sheet name` for the mappings specs for each form. So in this example, mappins for both of these forms should end up in the `collection.json`
![Image](https://github.com/user-attachments/assets/de13651e-fe9c-4534-9d43-b4d658ce52dd)

**Comment by AishaHassen** [comment]
@mtuchi @aleksa-krolls Here is the issue for sheet versioning

**Comment by aleksa-krolls** [comment]
@mtuchi this is ready to be worked on next. #1 was implemented on our call, so please make sure #2 works as we'd like as well. 

@AishaHassen I've removed this requirement (for now) because it adds a ton of complexity to the logic, and Mtuchi and I both think it might be way simpler to just overwrite the `collection.json` with whatever is in the Metadata.xls. Let's debate this requirement in-depth when you're back online. 
```
2. We want to use the **OMRS Form Version** to check if there was an update in the specific metadata sheet or not. 
     a. If there was an update in the version number(For example, if the form's initial version was v1.3 in the `collections.json` but is now 1.4) we will process and update the `collection.json` file. 
     b. If not we ignore the specific sheet.
```

**Comment by mtuchi** [comment]
@aleksa-krolls this ⏬ has been worked on already
>Make sure that for each "Active" sheet, refer to the column OMRS form sheet name for the mappings specs for each form. So in this example, mappings for both of these forms should end up in the collection.json

Search `F01-MHPSS Baseline v2` in this file https://github.com/OpenFn/openfn-lime-pilot/blob/next-staging/metadata/collections.json
You should see the collections for that form
